packages/salesforcedx-vscode-core/src/commands/commands.ts

```diff
@@ -48,6 +48,31 @@ export interface ParametersGatherer<T> {
   gather(): Promise<CancelResponse | ContinueResponse<T>>;
 }
 
+export class CompositeParametersGatherer<T> implements ParametersGatherer<T> {
```

Comment:
Composite class that will accept multiple ParameterGatherers and execute their gather() methods. If any of them return a CancelResponse then it will not proceed to the next gatherer (or executor if last gatherer).

I will look into the partial typing to replace the `any` types in another PR.